### ISSUE TITLE
InvalidOperationException: sub claim is missing

### ISSUE THREAD

**Issue by CrossBound** [IdentityServer/IdentityServer4#3461]
When attempting login using Windows Authentication, I keep getting an InvalidOperationException with message "sub claim is missing". 

In my login handler I'm explicitly setting the subject claim and passing it in to SignInAsync (shown below), but the SignInAsync handler is throwing the exception that the sub claim is missing.

This occurs _after_ the challenge and authentication. The Principal object is correctly being set in the handler prior to SignInAsync being called.

``` csharp
var newIdentity = new ClaimsIdentity("Windows");
newIdentity.AddClaim(new Claim(JwtClaimTypes.Subject, sid));
newIdentity.AddClaim(new Claim(JwtClaimTypes.Name, username));
var principal = new ClaimsPrincipal(newIdentity);

await HttpContext.SignInAsync(principal);
```
Below is a link to a minimal project that replicates the issue. 
https://github.com/CrossBound/WindowsIdentityServer

Per the issues linked below, I've already tried setting the sign-in scheme to "Windows" and making sure to clear the claim map prior to calling `AddIdentityServer()` but neither seems to help.

https://github.com/IdentityServer/IdentityServer4/issues/1939 
https://github.com/IdentityServer/IdentityServer4/issues/1707

**Comment by CrossBound** [comment]
I should note that the Privacy page in the client MVC app is the one that is protected so if you try to open that page it should trigger the login handler.

**Comment by leastprivilege** [comment]
Our host also has support for Windows authN - maybe compare your code with that.

**Comment by CrossBound** [comment]
I apologize for being obtuse, but I thought Windows authN is what I was doing.

I was basing my code on the information below, is there something else I should be looking at?

https://identityserver4.readthedocs.io/en/latest/topics/windows.html

https://github.com/IdentityServer/IdentityServer4.Quickstart.UI/blob/bedfed33462f4223bee4a0519c361c59aa8d3753/Quickstart/Account/ExternalController.cs#L148

**Comment by leastprivilege** [comment]
All I am saying is that our host also has Windows authentication and it is working. So maybe you want to check that to see if you spot any differences.

**Comment by CrossBound** [comment]
10-4, I will do that. When you say "our host" are you referring to the quickstart ui, the demo site, or something else?

**Comment by leastprivilege** [comment]
The host in he main repo

**Comment by CrossBound** [comment]
I have figured out the issue (I believe). One part of setting up HTTP.sys is adding `app.AddAuthentication("Windows");` in the `ConfigureServices` method. 

The problem was that I was adding the windows authentication handler _**after**_ I had already called ` JwtSecurityTokenHandler.DefaultInboundClaimTypeMap.Clear();` and the AddIdentity services.

Moving `app.AddAuthentication("Windows");` _**before**_ ` JwtSecurityTokenHandler.DefaultInboundClaimTypeMap.Clear();` solved the problem.

This is now working for me using HTTP.sys.

Thanks
